app/controllers/api/v1/salons_social_networks_controller.rb

```diff
@@ -0,0 +1,59 @@
+module Api
+  module V1
+    class SalonsSocialNetworksController < ApplicationController
+      before_action :set_salons_social_network, only: %i[update destroy]
+
+      # GET /salons/1/salons_social_networks
+      def index
+        salons_social_networks = SalonsSocialNetwork.where(salon_id: params[:salon_id]).all
+        render json: salons_social_networks
+      end
+
+      # GET /salons/1/salons_social_networks/1
+      def show
+        render json: set_salons_social_network
+      end
+
+      # POST /salons/1/salons_social_networks
+      def create
+        @salons_social_network = SalonsSocialNetwork.new(salons_social_network_params)
+
+        if @salons_social_network.save
+          render json: @salons_social_network
+        else
+          render json: { error: @salons_social_network.errors.full_messages }
+        end
+      end
+
+      # PUT /salons/1/salons_social_networks/1
+      def update
+        if set_salons_social_network.update(salons_social_network_params)
```

Comment:
You can use just `@salons_social_network` object because you are setting up in before_action helper